File: python/226-Invert Binary Tree.py
from bintree_helper import *

class Solution:
    def invertTree(self, root) -> TreeNode:
        "Time: O(n), Space: O(h)"
        if root is None: return root
        root.left, root.right = self.invertTree(root.right), self.invertTree(root.left)
        return root

root = deserialize("[4,2,7,1,3,6,9]")
result = Solution().invertTree(root)
drawtree(result)

File: cpp/226. Invert Binary Tree.cpp
#include <bits/stdc++.h>
using namespace std;
typedef long long ll;
typedef vector<int> vi;
typedef vector<vector<int>> vvi;
typedef pair<int, int> pi;


struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
// Time: O(n)
public:
    TreeNode* invertTree(TreeNode* root) {
        if(!root) return root;
        swap(root->left, root->right);
        invertTree(root->left);
        invertTree(root->right);
        return root;
    }
};

int main() {
    ios::sync_with_stdio(0); 
    cin.tie(0);

    
}